src/tacacs/bash_tacplus/bash_tacplus.c

```diff
@@ -350,40 +351,39 @@ int is_local_user(char *user)
     }
 
     struct passwd pwd;
-    struct passwd *pwdresult;
-    char *buf;
-    size_t bufsize = sysconf(_SC_GETPW_R_SIZE_MAX);
-    if (bufsize == -1) {
-        bufsize = DEFAULT_SC_GETPW_R_SIZE_MAX;
-    }
+    struct passwd *ppwd;
+    char buf[DEFAULT_GETPWENT_SIZE_MAX];
+    int pwdresult;
+    int result = ERROR_CHECK_LOCAL_USER;
+    setpwent();
+    while (1) {
+        pwdresult = getpwent_r(&pwd, buf, sizeof(buf), &ppwd);
+        if (pwdresult) {
+            // no more pw entry
+            break;
+        }
 
-    buf = malloc(bufsize);
-    if (buf == NULL) {
-       output_error("failed to allocate getpwnam_r buffer.\n");
-       return ERROR_CHECK_LOCAL_USER;
-    }
+        if (strcmp(ppwd->pw_name, user) != 0) {
+            continue;
```

Comment:
>continue [](http://example.com/codeflow?start=12&length=8)

This will lead to a loop, is it possible that the code flow always like this and keep burning CPU in an infinite loop?

Reply:
This will not be an infinity loop, the getpwent_r API designed to return next entry in passwd, the loop will end when there is no next entry:

https://www.man7.org/linux/man-pages/man3/getpwent_r.3.html